src/VASSAL/counters/BasicPiece.java

```diff
@@ -338,6 +373,21 @@ public void setProperty(Object key, Object val) {
     }
   }
 
+  @Override
+  public Command setPersistentProperty(Object key, Object newValue) {
+    if (persistentProps == null) {
+      persistentProps = new HashMap<>();
+    }
+
+    final Object oldValue = (newValue == null) ? persistentProps.remove(key) : persistentProps.put(key, newValue);
+    return Objects.equals(oldValue, newValue) ? null : new SetPersistentPropertyCommand (getId(), key, oldValue, newValue);
```

Comment:
Just checking to be sure this is what you want, because this surprised me when I first saw it: `Objects.equals()` returns `true` if both arguments are `null`.

Reply:
Yes, that is correct. We do not want to generate a Command if there has been no change in the value. If both are null, then no change, so return null for the result.